### PR TITLE
Average age

### PR DESCRIPTION
**Pull request by koffeinfrei**
Ready for review

### CODE REVIEW

File: lib/age.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Age
+  def self.from_date(date_of_birth)
+    now = Time.now.utc.to_date
+
+    leap_year_correction =
```
Comment (exul) [inline lib/age.rb:5]: :cop: Isn't this mainly to subtract one year from birth dates that are in the future of the current year? If so then the variable name is a bit misleading (although it's correct that this version also works for leap years).

**Comment by exul** [comment]
:+1:

File: lib/age.rb (hunk 1 above)
Comment (huerlisi) [inline lib/age.rb:5]: We can also do some more magical stuff:

```
relative = Time.at(Time.now - Time.parse(date_of_birth))
(relative.year - 1970) + relative.yday / 365.25
```

assumes you do not parse the date_of_birth as  Date in the caller.

:-)

Comment (huerlisi) [inline lib/age.rb:5]: Works also for date_of_birth before 1970, just checked;-)

File: lib/controllr.rb [hunk 2]
```diff
@@ -10,8 +11,19 @@ def contractor_count
...
+
+  def average_age
+    data = user_data.select { |user| user['employment'] == 'employee' }
+
+    ages = data.map { |user|
+      date_of_birth = Date.parse(user['date_of_birth'])
+      Age.from_date(date_of_birth)
+    }.compact
+
+    ages.inject(&:+) / ages.length
```
Comment (toupeira) [inline lib/controllr.rb:26]: `ages.sum` with ActiveSupport

File: lib/controllr.rb [hunk 3]
```diff
@@ -39,8 +56,9 @@ def fetch(api_url)
 
   def url(api_url)
     base = 'http://controllr.panter.biz'
-    token = "&user_token=#{ENV['CONTROLLR_TOKEN']}"
+    token = "user_token=#{ENV['CONTROLLR_TOKEN']}"
+    query_start = api_url.include?('?') ? '&' : '?'
 
-    "#{base}#{api_url}#{token}"
+    "#{base}#{api_url}#{query_start}#{token}"
```
Comment (toupeira) [inline lib/controllr.rb:62]: You can also use `URI.encode_www_form(hash)`

**Comment by toupeira** [comment]
:+1: looks good!

File: lib/controllr.rb (hunk 2 above)
Comment (koffeinfrei) [inline lib/controllr.rb:26]: we don't have activesupport, and I don't think we should add it for those few use cases we have.

Comment (toupeira) [inline lib/controllr.rb:26]: Agreed, I assumed it was already included through some other dependency :)

File: lib/controllr.rb (hunk 3 above)
Comment (koffeinfrei) [inline lib/controllr.rb:62]: Didn't know about that.

**Pull request merged**